src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> currentNode;
+        if (index == 0) {
+            if (head == null) {
+                Node<T> newNode = new Node<T>(null, value, null);
+                head = newNode;
+                tail = head;
+            } else {
+                currentNode = new Node<>(null, value, head);
+                head.prev = currentNode;
+                head = currentNode;
+            }
+        } else if (index == size) {
+            add(value);
+            return;
+        } else if (index == (size - 1)) {
+            if (checkIndex(index)) {
+                currentNode = new Node<>(tail.prev, value, tail);
+                tail.prev.next = currentNode;
+                tail.prev = currentNode;
+            }
+        } else {
+            currentNode = searchNodeByIndex(index);
+            Node<T> newNode = new Node<T>(currentNode.prev, value, currentNode);
+            newNode.prev.next = newNode;
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return searchNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode;
+        T e = null;
+        currentNode = searchNodeByIndex(index);
+        e = currentNode.item;
+        currentNode.item = value;
+        return e;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T deleteElement = null;
+        if (checkIndex(index)) {
+            deleteElement = (index == 0) ? unlinkFirst()
+                    : (index == size - 1) ? unlinkLast()
+                    : unlink(searchNodeByIndex(index));
+        }
+        return deleteElement;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = searchNodeByElement(object);
+        if (currentNode == null) {
+            return false;
+        } else if (currentNode.equals(head)) {
+            unlinkFirst();
+            return true;
+        } else if (currentNode.equals(tail)) {
+            unlinkLast();
+            return true;
+        }
+        currentNode.prev.next = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    public T unlink(Node<T> object) {
```

Comment:
Please do not make all service methods public. Only official logic should be public. All details of the realization have more strict modifiers.
Please recheck all methods